conans/client/source.py

```diff
@@ -19,7 +19,10 @@ def complete_recipe_sources(remote_manager, cache, conanfile, ref, remotes):
     occassions, conan needs to get them too, like if uploading to a server, to keep the recipes
     complete
     """
-    sources_folder = cache.package_layout(ref, conanfile.short_paths).export_sources()
+    short_paths = conanfile.python_requires[
```

Comment:
the fix is not here, but in the caller that calls ``complete_recipe_sources()``, which should pick the correct ``conanfile``. This can be agnostic of python_requires